CMakeLists.txt

```diff
@@ -1814,6 +1823,13 @@ if(HPX_WITH_DOCUMENTATION)
     PATTERN ".git" EXCLUDE)
 endif()
 
+
+if(HPX_WITH_VIM_YCM)
+  configure_file(${CMAKE_SOURCE_DIR}/tools/vim/.ycm_extra_conf.py
+               ${HPX_WITH_VIM_YCM}/.ycm_extra_conf.py @ONLY)
```

Comment:
I agree with you on that, it would be safer to have separate configuration files in build directories, but this is not how YCM works:
https://github.com/Valloric/YouCompleteMe#c-family-semantic-completion

> YCM looks for a .ycm_extra_conf.py file in the directory of the opened file or in any directory above it in the hierarchy (recursively); when the file is found, it is loaded (only once!) as a Python module.